test/vinyl-luatest/gh_6568_replica_initial_join_deletion_of_compacted_run_files_test.lua

```diff
@@ -0,0 +1,85 @@
+local cluster = require('test.luatest_helpers.cluster')
+local fio = require('fio')
+local t = require('luatest')
+
+local g = t.group('gh-6568-replica-initial-join-deletion-of-compacted-run-files')
+
+local s_id = 0
+g.before_all(function()
+    local helpers = require('test.luatest_helpers')
+
+    g.cluster = cluster:new({})
+
+    local master_uri = helpers.instance_uri('master')
+    local master_box_cfg = {
+        checkpoint_count = 1,
```

Comment:
No need to set `checkpoint_count` AFAIU.

Reply:
@locker following our test logic, during the replica's initial 'JOIN' stage, we want the snapshot fetched from master to be the one containing the inserted rows: otherwise, the replica will receive the boostrap snapshot, and only receive the 'insert' during the final 'JOIN' stage.

Reply:
Starting from commit 6332aca655ae7f95d391bdc0109e79915f6e6ad0, Tarantool sends data from a read view during initial join, not from the last snapshot, so whether you have a snapshot on the master or not doesn't matter, see

https://github.com/tarantool/tarantool/blob/e8765ba5cbd73ebb178459b008a9068ba25f98e2/src/box/vinyl.c#L3734-L3764

https://github.com/tarantool/tarantool/blob/e8765ba5cbd73ebb178459b008a9068ba25f98e2/src/box/vinyl.c#L3013-L3037

That is, in your case there will only be the initial join phase (no final join), no matter whether you create a snapshot or not; you just need to create a replica after you populated the test space.

Reply:
Fixed: thank you for pointing that out to me, did not know about the read view.